scripts/install_third_party_libs_test.py

```diff
@@ -44,41 +45,121 @@
+        class MockCheckCallReturn:
+            """Return object with required attributes."""
+
+            def __init__(self, returncode: int, message: bytes) -> None:
+                self.returncode = returncode
+                self.message = message
+            def communicate(self) -> Tuple[bytes, bytes]:
+                """Return required method."""
+                return b'', self.message
+        def mock_check_call(
+            _args: List[str],
+            **_kwargs: Union[int, None]
+        ) -> MockCheckCallReturn:
             self.check_function_calls['check_call_is_called'] = True
-            class Ret:
-                """Return object with required attributes."""
-
-                def __init__(self):
-                    self.returncode = 0
-                def communicate(self):
-                    """Return required method."""
-                    return b'', b''
-            return Ret()
-        def mock_check_call_error(*args, **kwargs) -> None:  # pylint: disable=unused-argument
+            return MockCheckCallReturn(0, b'')
+        def mock_check_call_error(
+                args: List[str], **_kwargs: Union[int, None]) -> None:
             """Raise the Exception resulting from a failed check_call()"""
             self.check_function_calls['check_call_is_called'] = True
             raise subprocess.CalledProcessError(-1, args[0])
-        def mock_popen_error_call(unused_cmd_tokens, *args, **kwargs):  # pylint: disable=unused-argument
-            class Ret:
-                """Return object that gives user-prefix error."""
-
-                def __init__(self):
-                    self.returncode = 1
-                def communicate(self):
-                    """Return user-prefix error as stderr."""
-                    return b'', b'can\'t combine user with prefix'
-            return Ret()
-        def mock_print(msg, end=''):  # pylint: disable=unused-argument
-            self.print_arr.append(msg)
+        def mock_popen_error_call(
+            _args: List[str],
+            **_kwargs: Union[int, None]
+        ) -> MockCheckCallReturn:
+            return MockCheckCallReturn(
+                1, b'can\'t combine user with prefix')
+        def mock_ensure_directory_exists(_path: str) -> None:
+            pass
+        class MockZipFile(zipfile.ZipFile):
+            def __init__(self, _path: str, _mode: str) -> None:  # pylint: disable=super-init-not-called
+                pass
+        def mock_extractall(_file: zipfile.ZipFile, **_kwargs: str) -> None:
+            self.check_function_calls['extractall_is_called'] = True
+        def mock_extractall_raises_exception(
+                _file: zipfile.ZipFile, **_kwargs: str) -> None:
+            self.check_function_calls['extractall_is_called'] = True
+            raise Exception()
+        def mock_isfile_true(_filename: str) -> bool:
+            return True
+        def mock_isfile_false(_filename: str) -> bool:
```

Comment:
```suggestion
        def mock_isfile_false(_filename: str) -> Literal[False]:
```

ditto elsewhere

Reply:
Bit confused here: wasn't Literal only added to typing in Python3.8, while Oppia uses 3.7? Is there a way I can access this through `__future__` that I'm missing?

Reply:
Done.